torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -102,135 +112,256 @@ def get_all_unique_feature_names(self, plottable_features_only: bool = True) ->
+            feature_filter (str): The feature filter, if we want to filter the set of data to only include
+                a certain set of features that include feature_filter
+                If feature = "", then we do not filter based on any features
+            module_fqn_filter (str): The filter on prefix for the module fqn. All modules that have fqn with
+                this prefix will be included
+                If module_fqn_filter = "" we do not filter based on module fqn, and include all modules
+
+        First, the data is filtered based on module_fqn, and then filtered based on feature
+        Returns an OrderedDict (sorted in order of model) mapping:
+            module_fqns -> feature_names -> values
+        """
+        # create return dict
+        filtered_dict: OrderedDict[str, Any] = OrderedDict()
 
-        Table columns:
+        for module_fqn in self.generated_reports:
+            # first filter based on module
+            if module_fqn_filter == "" or module_fqn_filter in module_fqn:
+                # create entry for module and loop through features
+                filtered_dict[module_fqn] = {}
+                module_reports = self.generated_reports[module_fqn]
+                for feature_name in module_reports:
+                    # check if filtering on features and do so if desired
+                    if feature_filter == "" or feature_filter in feature_name:
+                        filtered_dict[module_fqn][feature_name] = module_reports[feature_name]
 
-         idx  layer_fqn   type  shape  feature_1   feature_2   feature_3   .... feature_n
-        ----  ---------   ----  -----  ---------   ---------   ---------        ---------
+        # we have populated the filtered dict, and must return it
 
-        Args:
-            feature (str, optional): The specific feature we wish to generate the table for
-                Default = "", results in all the features being printed
-            module_fqn_prefix_filter (str, optional): Only includes modules with this string prefix
-                Default = "", results in all the modules in the reports to be visible in the table
-
-        Returns a tuple with two objects:
-            (str) The formatted string that contains the table information to be printed
-            (List[List[Any]]) A list of lists containing the table information row by row
-                The 0th index row will contain the headers of the columns
-                The rest of the rows will contain data
-        Expected Use:
-            >>> tabluated_str, info = model_report_visualizer.generate_table_info(*filters)
-            >>> print(tabulated_str) # outputs neatly formatted table
-        """
-        pass
+        return filtered_dict
 
-    def generate_table_visualization(self, feature: str = "", module_fqn_prefix_filter: str = ""):
+    def generate_filtered_tables(self, feature_filter: str = "", module_fqn_filter: str = "") -> Dict[str, List[List[Any]]]:
```

Comment:
this method is pretty long, can we split it into multiple smaller functions?

Reply:
broke this up by adding two helper functions, each one is much smaller now